lib/utils.js

```diff
@@ -19,17 +19,25 @@ function isRedisReady(client) {
       resolve();
     } else {
       function handleReady() {
+        client.removeListener('end', handleEnd);
         client.removeListener('error', handleError);
         resolve();
       }
 
+      let lastError;
       function handleError(err) {
+        lastError = err;
+      }
+
+      function handleEnd() {
         client.removeListener('ready', handleReady);
-        reject(err);
+        client.removeListener('error', handleError);
+        reject(lastError);
```

Comment:
What if the queue was closed manually?,wouldn't it also reject this promise? Maybe it should only reject if lastError is not undefined, otherwise just resolve.

Reply:
Makes sense. I'll check and let you know.

Reply:
I'm thinking it should reject still, even when the connection is manually closed.
Considering this is a very particular case where the 'end' listener will be called if the connection never got to a 'ready' state in the first place.
If connection ends, then pending initialization should still fail I think, since the connection was never established. Otherwise, anything waiting on this, after resolved, will try to send commands to Redis on a closed connection.
What do you think ?

Reply:
yeah, I guess it is safer to reject to avoid other corner cases, although some people will complain that they get errors when closing too fast in other corner cases :)